src/utils/ShuffleGraph.java

```diff
@@ -334,17 +370,37 @@ public boolean retainAll(Collection<?> c) {
     }
 
     protected class GraphIterator implements Iterator<ShuffleInfo> {
```

Comment:
these are nested classes, but I feel also prefixing them with "shuffle" would be good for consistency (for class names) and to prevent conflicts within the class?

Reply:
They are protected, so that's unnecessary. I used a similar naming convention similar to that used in the source [code for `ArrayList`](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/util/ArrayList.java#L982) (at least it's not called just `Itr` lol).

Reply:
the point of my message was for conflicts *within* the class, though (but yeah, it doesn't matter much)